src/main/java/org/mondemand/Client.java

```diff
@@ -534,9 +534,6 @@ public void flush(boolean resetStats) {
         contextStats.clear();
       }
     }
-    if(samples != null) {
-      samples.clear();
-    }
```

Comment:
dispatchStatsSamples swaps out the map already which is called earlier so no need to clear.